app-modules/basic-needs/src/Filament/Actions/SendEmailAction.php

```diff
@@ -0,0 +1,359 @@
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\BasicNeeds\Filament\Actions;
+
+use AdvisingApp\Engagement\Actions\CreateEngagement;
+use AdvisingApp\Engagement\DataTransferObjects\EngagementCreationData;
+use AdvisingApp\Engagement\Models\Engagement;
+use AdvisingApp\Notification\Enums\NotificationChannel;
+use AdvisingApp\Prospect\Models\Prospect;
+use AdvisingApp\Prospect\Models\ProspectEmailAddress;
+use AdvisingApp\StudentDataModel\Models\Student;
+use AdvisingApp\StudentDataModel\Models\StudentEmailAddress;
+use Filament\Actions\Action;
+use Filament\Forms\Components\DateTimePicker;
+use Filament\Forms\Components\Select;
+use Filament\Forms\Components\Toggle;
+use Filament\Pages\Page;
+use Filament\Schemas\Components\Grid;
+use Filament\Schemas\Components\Utilities\Get;
+use Filament\Schemas\Components\Utilities\Set;
+use Filament\Schemas\Components\Wizard\Step;
+use Filament\Schemas\Schema;
+use FilamentTiptapEditor\Enums\TiptapOutput;
+use FilamentTiptapEditor\TiptapEditor;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Query\Expression;
+use Illuminate\Support\Carbon;
+use Livewire\Features\SupportFileUploads\TemporaryUploadedFile;
+
+class SendEmailAction extends Action
+{
+    protected Student|Prospect|null $educatable = null;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->icon('heroicon-m-chat-bubble-bottom-center-text')
+            ->modalHeading('Send Engagement')
+            ->model(Engagement::class)
+            ->authorize(function () {
+                return auth()->user()->can('create', Engagement::class);
+            })
+            ->steps(fn (): array => [
+                Step::make('Contact Information')
+                    ->schema([
+                        Select::make('recipient_type')
+                            ->label('Recipient Type')
+                            ->options([
+                                'student' => 'Student',
+                                'prospect' => 'Prospect',
+                            ])
+                            ->live()
+                            ->afterStateUpdated(fn (Set $set) => $set('recipient_id', null))
+                            ->required(),
+                        Select::make('recipient_id')
```

Comment:
I think you should hide this field until the `type` is filled, otherwise it looks empty